neural_speed/models/baichuan/baichuan.cpp

```diff
@@ -106,6 +106,13 @@ static bool baichuan_model_eval_internal(model_context* ctx, const model_input*
   // otherwise, the threads are spin-lock waiting for the BLAS calls and are degrading the performance
   ne_cgraph gf = {};
   gf.n_threads = N >= 32 && ne_cpu_has_blas() ? 1 : n_threads;
+  if (hparams.ftype == NE_FTYPE_MOSTLY_NF4 && bestla_is_hybrid() && gf.n_threads > bestla_get_Pcore_number()) {
+    printf(
+        "WARNING: NF4 is poor at Ecore, only use P-core to inference. Not use the thread number according to settings. "
+        "New thread number is: %d\n.",
+        bestla_get_Pcore_number());
+    gf.n_threads = bestla_get_Pcore_number();
+  }
```

Comment:
I will prefer to subtract out these logic so you can keep improving within bestla

```cpp
// ne_bestla.cpp
int bestla_get_best_thread_num(ne_ftype ft){
  if (ftype == NE_FTYPE_MOSTLY_NF4 && bestla_is_hybrid()) return bestla_get_Pcore_number();
  else return ${max_thread_num}
}

// various models.cpp
int best_n_thread = bestla_get_best_thread_num(hparams.ftype);
 gf.n_threads = N >= 32 && ne_cpu_has_blas() ? 1 : (best_n_thread < n_threads : best_n_threads : n_threads);

Reply:
Done